src/main/kotlin/org/rust/ide/actions/macroExpansion/macroExpansionViewUtils.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions.macroExpansion
+
+import com.intellij.ide.highlighter.HighlighterFactory
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.editor.ex.EditorEx
+import com.intellij.openapi.editor.highlighter.EditorHighlighter
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.ui.popup.PopupPositionManager
+import org.rust.lang.core.macros.RsExpandedElement
+import org.rust.lang.core.macros.parseExpandedTextWithContext
+import org.rust.lang.core.psi.RsMacroCall
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.ext.expandAllMacrosRecursively
+import org.rust.lang.core.psi.ext.expansion
+import java.awt.BorderLayout
+import javax.swing.JPanel
+
+/**
+ * Helper class to expand macro (recursively or not, depending on [expandRecursively]), and show it in the floating
+ * window using [MacroExpansionViewComponent].
+ */
+class MacroExpansionViewCreator(private val expandRecursively: Boolean) {
+
+    fun expandAndShowMacro(project: Project, editor: Editor, macroToExpand: RsMacroCall) {
+        val expansions = expandMacroCall(macroToExpand) ?: return
+
+        if (expansions.isEmpty()) return
+
+        val component = MacroExpansionViewComponent(expansions)
+
+        val createPopup = JBPopupFactory.getInstance().createComponentPopupBuilder(component, component)
```

Comment:
Maybe rename `createPopup` to `popup`?

Reply:
:+1: or convert to a function.
It's weird to name properties as a verb